spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -114,11 +117,22 @@
   private final Long snapshotId;
   private final boolean refreshEagerly;
   private final Set<TableCapability> capabilities;
+  private String branch;
   private StructType lazyTableSchema = null;
   private SparkSession lazySpark = null;
 
   public SparkTable(Table icebergTable, boolean refreshEagerly) {
-    this(icebergTable, null, refreshEagerly);
+    this(icebergTable, (Long) null, refreshEagerly);
+  }
+
+  public SparkTable(Table icebergTable, String branch, boolean refreshEagerly) {
+    this(icebergTable, refreshEagerly);
+    this.branch = branch;
+    ValidationException.check(
+        branch != null
+            && (SnapshotRef.MAIN_BRANCH.equals(branch) || table().snapshot(branch) != null),
+        "Cannot operate against non-existing branch: %s",
```

Comment:
I think it should be "non-existent" instead of "non-existing", but both are kind of hard to understand. Maybe it would be better to say `"Cannot use branch (does not exist): %s"` instead?

Reply:
updated!